mathesar_ui/src/components/modal/Modal.svelte

```diff
@@ -0,0 +1,26 @@
+<script lang="ts">
+  import Portal from 'svelte-portal/src/Portal.svelte';
+
+  export let overlayTarget = 'body';
+
+  // Additional classes
+  let classes = '';
+  export { classes as class };
+
+  // Inline styles
+  export let style = '';
+</script>
+
+<div class="modal-wrapper">
+  <div class={['modal', classes].join(' ')} {style}>
+    <slot/>
+  </div>
+</div>
+
+<Portal target={overlayTarget}>
+  <div class="overlay"></div>
+</Portal>
```

Comment:
We already have a `portal` action. We do not require the `svelte-portal` package. We could do this:

```suggestion
<div class="overlay" use:portal={{ target: overlayTarget  }}/>
```

Reply:
Replaced the package with the portal action in 017fc5b.

Unfortunately Storybook complains when importing from `@mathesar-components` so I had to turn off the story for the modal.

Reply:
@dhruvkb Could you please open a bug report for storybook not accepting aliases?

Reply:
Filed bug report #528 to address this.